extensions/wikia/MercuryApi/helpers/MercuryApiCategoryCacheHelper.class.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * We use this class to easily invalidate all category objects in cache related to a particular
+ * page. The way to go is to call getTouched() and incorporate the resulting number to
+ * the key used for getting the actual value from cache.
+ *
+ * On modification call setTouched to cause the cache to be invalidated.
+ *
+ * FIXME this is unnecessarily complicated
+ * It would be better (simpler) to cache category data on Varnish and use surrogate keys to purge the responses.
+ * We can't do this now, because Mercury uses icache to get the data from API.
+ * Our icache doesn't support the surrogate keys at this point of time, see OPS-10115
+ */
+class MercuryApiCategoryCacheHelper {
+
+	const CACHE_TTL = 60 * 60 * 24;
+	const CACHE_VERSION = 1;
+
+	public static function getTouched( string $categoryDBKey ) {
+		global $wgMemc;
+		return $wgMemc->get( self::getTouchedKey( $categoryDBKey ) );
+	}
+
+	public static function setTouched( string $categoryDBKey ) {
+		global $wgMemc;
+		$wgMemc->set( self::getTouchedKey( $categoryDBKey ), time() . rand( 0, 9999 ), self::CACHE_TTL );
```

Comment:
It's copied from https://github.com/Wikia/app/blob/87bf310b4201056162406313d186cec57d6c6e90/extensions/wikia/CategoryExhibition/CategoryExhibitionCacheHelper.class.php#L20-L20 and this whole file will be removed after icache surrogate keys work.